Core/NoOpRunner.Core/Shapes/BaseShape.cs

```diff
@@ -153,9 +154,20 @@ public void ShiftShapes()
 
         public List<T> GetOfType<T>() where T : IMapPart
         {
-            Logging.Instance.Write($"[Composite/{nameof(BaseShape)}] {nameof(GetOfType)}", LoggingLevel.CompositePattern);
+            Logging.Instance.Write($"[Composite/{nameof(BaseShape)}] {nameof(GetOfType)}", LoggingLevel.Composite);
 
             return new List<T>();
         }
+
+        public void Accept(INodeVisitor visitor)
+        {
+            if (this is StaticShape shape)
```

Comment:
Shouldn't this be an overloaded method in each of the child classes? The base class shouldn't know anything about the existance of it's descendants.

Reply:
Oops, fair point